packages/amplify-console-integration/__tests__/pullAndInit.test.ts

```diff
@@ -0,0 +1,156 @@
+        useProfile: true,
+        profileName: util.getProfileName(),
+      },
+    };
+    const codegenParam = {
+      generateCode: false,
+      generateDocs: false,
+    };
+
+    let teamProviderInfo;
+    //test for init a clean frontend project
+    const projectDirPath = util.createNewProjectDir();
+    await headlessInit(projectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(projectDirPath)).toBeTruthy();
+    teamProviderInfo = getTeamProviderInfo(projectDirPath);
+    expect(teamProviderInfo).toBeDefined();
+
+    //test for existing env
+    removeFilesForTeam(projectDirPath);
+    await headlessPull(projectDirPath, amplifyParam, providersParam);
+    await headlessInit(projectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(projectDirPath)).toBeTruthy();
+    teamProviderInfo = getTeamProviderInfo(projectDirPath);
+    expect(teamProviderInfo).toBeDefined();
+    expect(teamProviderInfo['enva']).toBeDefined();
+
+    //test for new env
+    envName = 'envb';
+    backendParams = generateBackendEnvParams(appId, projectName, envName);
+
+    await createBackendEnvironment(backendParams, amplifyClient);
+
+    removeFilesForTeam(projectDirPath);
+    amplifyParam = {
+      envName,
+      appId,
+    };
+    await headlessInit(projectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(projectDirPath)).toBeTruthy();
+    teamProviderInfo = getTeamProviderInfo(projectDirPath);
+    expect(teamProviderInfo).toBeDefined();
+    expect(teamProviderInfo['envb']).toBeDefined();
+
+    //clean up after the tests
+    await headlessDelete(projectDirPath);
+    util.deleteProjectDir(projectDirPath);
+  });
+
+  it('connects to third party project', async () => {
+    const amplifyClient = getConfiguredAmplifyClient();
+
+    const projectName = 'cliintegrationthirdparty';
+    let envName = 'devteama';
+    let appId = await createConsoleApp(projectName, amplifyClient);
+    let backendParams = generateBackendEnvParams(appId, projectName, envName);
+    await createBackendEnvironment(backendParams, amplifyClient);
+
+    let amplifyParam = {
+      envName,
+      appId,
+    };
+    const providersParam = {
+      awscloudformation: {
+        configLevel: 'project',
+        useProfile: true,
+        profileName: 'default',
+      },
+    };
+    const codegenParam = {
+      generateCode: false,
+      generateDocs: false,
+    };
+
+    //create the original project
+    const originalProjectDirPath = util.createNewProjectDir('original');
+    await headlessInit(originalProjectDirPath, amplifyParam, providersParam, codegenParam);
+    expect(checkAmplifyFolderStructure(originalProjectDirPath)).toBeTruthy();
+    let originalTeamProviderInfo = getTeamProviderInfo(originalProjectDirPath);
+    expect(originalTeamProviderInfo).toBeDefined();
+    expect(originalTeamProviderInfo['devteama']).toBeDefined();
```

Comment:
@UnleashedMind Same

```suggestion
    expect(teamProviderInfo[envName]).toBeDefined();
```

Reply:
We have two teams, the original team and the team that clones the original team's github repo, and set up their own backend resources. 
and since the two provider info objects are in the same scope, we have to distinguish them, hence one is named "originalTeamProviderInfo" and the other is named "clonedTeamProviderInfo"